nnpdf_data/nnpdf_data/theorydbutils.py

```diff
@@ -51,7 +51,17 @@ def fetch_theory(theory_database: Path, theoryID: int):
     >>> from nnpdf_data.theorydbutils import fetch_theory
     >>> theory = fetch_theory(theory_cards, 700)
     """
-    filepath = theory_database / f"{theoryID}.yaml"
+
+    # Since theoryfile names may contain underscores, the theoryfile name cannot uniquely be
+    # determined from the theoryID. Therefore we create a mapping between the theoryid as python
+    # int and the corresponding file
+    available_theories = {
+        int(theoryfile.name.removesuffix('.yaml')): theoryfile.name
+        for theoryfile in theory_database.glob("*.yaml")
+    }
```

Comment:
I would cover this as well with a try-except block in case someone left a file "not_a_theory.yaml" in this folder with "file whatever doesn't conform to the .yaml convention"

It would be also good to check that not two files are the same.

Something like:
```suggestion
    available_theories = {}
    for theoryfile in theory_database.glob("*.yaml"):
        try:
            tmp_id = int(theoryfile.name.removesuffix('.yaml'))
        except ValueError as e:
             return InvalidTheoryName(f"Bad theory name for{theoryfile}") from e
         if tmp_id in available_theories:
             another = available_theories[tmp_id]
             raise ValueError("Two theory files with same id: {theoryfile} and {another}")
         available_theories[tmp_id] = theoryfile
```

And I would make this into its own function `get_available_theory_cards(path)` with a `lru_cache` so that for a given path you don't do all this every time but then a test that simply loops over all the theories would catch a few possible bugs that the underscores may create (for instance, having theory 400.yaml and 40_0.yaml and 4_00.yaml)